app/src/main/java/de/westnordost/streetcomplete/map/PinIcons.kt

```diff
@@ -59,5 +60,20 @@ import de.westnordost.streetcomplete.R
     when (map["tourism"]) {
         "information" -> return R.drawable.ic_pin_information
     }
+    if ((map.containsKey("addr:housenumber") || map.containsKey("addr:housename"))
+        && !map.containsAnyKey("amenity", "craft", "leisure", "office", "shop", "tourism")){
+        return R.drawable.ic_none
+    }
     return null
 }
+
+fun getTitle(map: Map<String, String>): String? {
+    return if (map["name"] != null || map["brand"] != null) {
+        map["name"] ?: map["brand"]
+    }
+    else if (map["addr:conscriptionnumber"] != null && map["streetnumber"] != null) {
```

Comment:
```suggestion
    } else if (map["addr:conscriptionnumber"] != null && map["streetnumber"] != null) {
```

Reads odd otherwise .oO("how can this still be executed if there is already a return?")

Reply:
Also, you mean `addr:streetnumber`, not `streetnumber`.